crates/proto/src/dnssec/rdata/cdnskey.rs

```diff
@@ -7,41 +7,169 @@
 
 //! CDNSKEY type and related implementations
 
-use std::{fmt, ops::Deref};
+use std::fmt;
 
 #[cfg(feature = "serde")]
 use serde::{Deserialize, Serialize};
 
 use crate::{
+    dnssec::{Algorithm, PublicKeyBuf},
     error::ProtoResult,
     rr::{RData, RecordData, RecordDataDecodable, RecordType},
-    serialize::binary::{BinDecoder, BinEncodable, BinEncoder, Restrict},
+    serialize::binary::{BinDecoder, BinEncodable, BinEncoder, Restrict, RestrictedMath},
+    ProtoError, ProtoErrorKind,
 };
 
-use super::{DNSSECRData, DNSKEY};
+use super::DNSSECRData;
 
 /// Child DNSKEY. See RFC 8078.
 #[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
 #[derive(Debug, PartialEq, Eq, Hash, Clone)]
-pub struct CDNSKEY(DNSKEY);
+pub struct CDNSKEY {
```

Comment:
This commit makes it not that clear why this change is happening. Suggest "inlining" `DNSKEY` directly first and then adding another commit that makes the necessary changes.